### PR TITLE
rbd-nbd: support optionally setting device timeout

### PR DESCRIPTION
**Pull request by trociny**
Fixes: http://tracker.ceph.com/issues/22333
Signed-off-by: Mykola Golub <to.my.trociny@gmail.com>

### CODE REVIEW

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 1]
```diff
@@ -85,6 +86,7 @@ static void usage()
             << "  --nbds_max <limit>      Override for module param nbds_max\n"
             << "  --max_part <limit>      Override for module param max_part\n"
             << "  --exclusive             Forbid writes by other clients\n"
+            << "  --timeout <seconds>     Set nbd request timeout\n"
```
Comment (tchaikov) [inline src/tools/rbd_nbd/rbd-nbd.cc]:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/rbd/help.t: failed
--- /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/rbd/help.t
+++ /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/rbd/help.t.err
@@ -1148,6 +1148,7 @@
   usage: rbd nbd map [--pool <pool>] [--image <image>] [--snap <snap>] 
                      [--read-only] [--exclusive] [--device <device>] 
                      [--nbds_max <nbds_max>] [--max_part <max_part>] 
+                     [--timeout <timeout>] 
                      <image-or-snap-spec> 
   
   Map image to a nbd device.
@@ -1165,6 +1166,7 @@
     --device arg          specify nbd device
     --nbds_max arg        override module param nbds_max
     --max_part arg        override module param max_part
+    --timeout arg         set nbd request timeout (seconds)
   
   rbd help nbd unmap
   usage: rbd nbd unmap 
...
The following tests FAILED:
	  2 - run-cli-tests (Failed)
```

this breaks `run-cli-tests`

Comment (trociny) [reply src/tools/rbd_nbd/rbd-nbd.cc]: @tchaikov updated. Thanks

**Comment by ldidry** [comment]
This has been merged to master in december, but I don't see it in new releases (nor v12.2.3 nor v13.0.1, both from less than a month ago).

Shouldn't it be backported to v12 and v13 if possible? (I would have great use of it and I use v12)

**Comment by ldidry** [comment]
Thank you!

The redmine ticket didn't have any backport nor target version, so I thought that it would be in all last versions. Now I'll know that it just means "target: last version"

**Comment by dillaman** [comment]
We generally try to restrict features to new releases and only backport bug fixes to lessen the load on our volunteer backporting team (and to reduce risks of regressions on stable releases). The next mimic dev snapshot release will be cut this week or next and will include the change.

**Pull request merged**